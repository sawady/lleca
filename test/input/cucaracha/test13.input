
fun f(h : Int, o : Int, l : Int, a : Int, enter : Int) {
  h := 104
  o := 111
  l := 108
  a := 97
  putChar(h)
  putChar(o)
  putChar(l)
  putChar(a)
  putChar(enter)
}

fun main() {
  f(2, 3, 4, 5, 10)
  f(22, 33, 44, 55, 10)
}
