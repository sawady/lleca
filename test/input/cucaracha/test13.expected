Cons(
  Function(
    f,
    Cons(
      Parameter(
        h,
        Int
      ),
      Cons(
        Parameter(
          o,
          Int
        ),
        Cons(
          Parameter(
            l,
            Int
          ),
          Cons(
            Parameter(
              a,
              Int
            ),
            Cons(
              Parameter(
                enter,
                Int
              ),
              Nil
            )
          )
        )
      )
    ),
    Unit,
    Cons(
      StmtAssign(
        h,
        ExprConstNum(
          104
        )
      ),
      Cons(
        StmtAssign(
          o,
          ExprConstNum(
            111
          )
        ),
        Cons(
          StmtAssign(
            l,
            ExprConstNum(
              108
            )
          ),
          Cons(
            StmtAssign(
              a,
              ExprConstNum(
                97
              )
            ),
            Cons(
              StmtCall(
                putChar,
                Cons(
                  ExprVar(
                    h
                  ),
                  Nil
                )
              ),
              Cons(
                StmtCall(
                  putChar,
                  Cons(
                    ExprVar(
                      o
                    ),
                    Nil
                  )
                ),
                Cons(
                  StmtCall(
                    putChar,
                    Cons(
                      ExprVar(
                        l
                      ),
                      Nil
                    )
                  ),
                  Cons(
                    StmtCall(
                      putChar,
                      Cons(
                        ExprVar(
                          a
                        ),
                        Nil
                      )
                    ),
                    Cons(
                      StmtCall(
                        putChar,
                        Cons(
                          ExprVar(
                            enter
                          ),
                          Nil
                        )
                      ),
                      Nil
                    )
                  )
                )
              )
            )
          )
        )
      )
    )
  ),
  Cons(
    Function(
      main,
      Nil,
      Unit,
      Cons(
        StmtCall(
          f,
          Cons(
            ExprConstNum(
              2
            ),
            Cons(
              ExprConstNum(
                3
              ),
              Cons(
                ExprConstNum(
                  4
                ),
                Cons(
                  ExprConstNum(
                    5
                  ),
                  Cons(
                    ExprConstNum(
                      10
                    ),
                    Nil
                  )
                )
              )
            )
          )
        ),
        Cons(
          StmtCall(
            f,
            Cons(
              ExprConstNum(
                22
              ),
              Cons(
                ExprConstNum(
                  33
                ),
                Cons(
                  ExprConstNum(
                    44
                  ),
                  Cons(
                    ExprConstNum(
                      55
                    ),
                    Cons(
                      ExprConstNum(
                        10
                      ),
                      Nil
                    )
                  )
                )
              )
            )
          ),
          Nil
        )
      )
    ),
    Nil
  )
)
